### DISCUSSION TITLE
Load model saved with pytorch lightning without pytorch lightning installed?

### DISCUSSION THREAD

**Discussion by radiobotics-djernaes** [Lightning-AI/pytorch-lightning#7059] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Hey, 

as the title says. Is it possible to load the pytorch-lightning .ckpt file without using pytorch lightning? I would prefer to not have pytorch lightning installed in my inference .env and was wondering, if I could load it using purely pytorch?

**Comment by awaelchli** [comment]
Sadly, the answer currently is: no, not always. 
I believe you can if there are not callbacks saved to the checkpoint (e.g. ModelCheckpoint). 
The problem is that if objects from the Lightning API get pickled, one needs to have lightning in the environment to load (unpickle) them. 

We are working on improving this so that these checkpoints can be loaded independently of Lightning! It's a priority for us after 1.3!
Progress here: #6886

**Reply by ananthsub** [reply]
Do you need the full checkpoint for inference? Are you able to extract the model(s) you're interested in and get their state dicts? Alternatively for inference you could use the to_torchscript hook to get a ScriptModule you can use for inference
